src/test/java/com/codeus/winter/config/PackageBeanRegistrationTest.java

```diff
@@ -30,65 +27,59 @@ class PackageBeanRegistrationTest {
     private PackageBeanRegistration beanRegistration;
 
     /**
-     * Sets up the test environment by mocking dependencies and initializing
-     * the {@link PackageBeanRegistration} instance.
+     * Sets up the test environment by mocking dependencies and initializing the
+     * {@link PackageBeanRegistration} instance.
      */
     @BeforeEach
     void setUp() {
-        packageScanner = mock(PackageScanner.class);
-        registry = mock(BeanDefinitionRegistry.class);
+        packageScanner = new PackageScannerImpl();
+        registry = new BeanDefinitionRegistryImpl();
         beanRegistration = new PackageBeanRegistration(packageScanner, registry);
     }
 
-    /**
-     * Tests that classes annotated with {@link Component} are successfully
-     * registered as {@link BeanDefinition}s in the registry.
-     */
     @Test
     void shouldRegisterBeansSuccessfully() {
-        Class<?> testClass = WinterComponent.class;
-        when(packageScanner.findClassesWithAnnotations(anyString(), anySet()))
-                .thenReturn(Set.of(testClass));
-        when(registry.containsBeanDefinition("winterComponent")).thenReturn(false);
-        beanRegistration.registerBeans("com.framework");
+        beanRegistration.registerBeans("com.codeus.winter.config");
 
-        verify(registry).registerBeanDefinition(eq("winterComponent"), any(BeanDefinitionImpl.class));
+        BeanDefinition definition = registry.getBeanDefinition("winterComponent");
+        assertNotNull(definition);
+        // bean name
+        assertEquals("com.codeus.winter.config.WinterComponent", definition.getBeanClassName());
+        // scope
+        assertEquals(SCOPE_PROTOTYPE, definition.getScope());
+        assertFalse(definition.isSingleton());
+        // depends on
+        String[] dependencies = {"com.codeus.winter.config.QualifierComponent",
+            "com.codeus.winter.config.AutowiredComponent"};
+        assertTrue(Arrays.equals(dependencies, definition.getDependsOn()));
+        // init method name
+        assertEquals("init", definition.getInitMethodName());
+        // destroy method name
+        assertEquals("destroy", definition.getDestroyMethodName());
     }
 
     /**
-     * Tests that a {@link NotUniqueBeanDefinitionException} is thrown if two
-     * classes with the same bean name are registered.
+     * Tests that a {@link NotUniqueBeanDefinitionException} is thrown if two classes with the same
+     * bean name are registered.
      */
     @Test
     void shouldThrowExceptionForDuplicateBeanName() {
```

Comment:
In addition to this test case, please add one more: make two test classes with the same name and place them into two different packages. Use a higher-level package (including both packages) in the `beanRegistration.registerBeans` call and assert the exception.

Reply:
Great. But the 2nd `DuplicateComponent` looks a bit weird among test classes, outside test fixture package. 
Let's create another package within `com.codeus.winter.config.test` - for example `inner` and move the 2nd  `DuplicateComponent` there. So, the final picture looks like:
- `com.codeus.winter.config.test.DuplicateComponent` - 1st one.
- `com.codeus.winter.config.test.inner.DuplicateComponent` - 2nd one.

Note, `shouldThrowExceptionForDuplicateBeanNamePassedSamePackageTwoTimes` test case should be adjusted. I believe you can easily reuse that new package for it `com.codeus.winter.config.test.inner`.

Reply:
done